custom_components/tuya_local/devices/unknow_2way_energymeter.yaml

```diff
@@ -0,0 +1,452 @@
+        step: 1
+        range:
+          min: 800
+          max: 1200
+        mapping:
+          - scale: 1000
+  - entity: sensor
+    class: frequency
+    category: diagnostic
+    dps:
+      - id: 102
+        type: integer
+        name: sensor
+        unit: Hz
+        class: measurement
+        mapping:
+          - scale: 100
+      - id: 128
+        type: integer
+        name: calibration
+        step: 1
+        range:
+          min: 800
+          max: 1200
+        mapping:
+          - scale: 1000
+  - entity: sensor
+    name: Total power
+    class: power
+    category: diagnostic
+    dps:
+      - id: 103
+        type: integer
+        name: sensor
+        unit: W
+        class: measurement
+        mapping:
+          - scale: 10
+  - entity: sensor
+    name: Power factor A
+    class: power_factor
+    category: diagnostic
+    dps:
+      - id: 104
+        type: integer
+        name: sensor
+        unit: "%"
+        class: measurement
+        mapping:
+          - scale: 100
+  - entity: sensor
+    name: Current A
+    class: current
+    category: diagnostic
+    dps:
+      - id: 105
+        type: integer
+        name: sensor
+        unit: A
+        class: measurement
+        mapping:
+          - scale: 1000
+      # - id: 129
+      #   type: integer
+      #   name: calibration
+      #   step: 1
+      #     range:
+      #       min: 800
+      #       max: 1200
+      #     mapping:
+      #       - scale: 1000
+  - entity: sensor
+    name: Power A
+    class: power
+    category: diagnostic
+    dps:
+      - id: 106
+        type: integer
+        name: sensor
+        unit: W
+        class: measurement
+        mapping:
+          - scale: 10
+      - id: 124
+        type: boolean
+        name: direction
+  #     - id: 130
+  #       type: integer
+  #       name: calibration
+  #       step: 1
+  #       range:
+  #         min: 800
+  #         max: 1200
+  #       mapping:
+  #         - scale: 1000
+  - entity: sensor
+    name: Forward energy A
+    class: energy
+    category: diagnostic
+    dps:
+      - id: 107
+        type: integer
+        name: sensor
+        unit: kWh
+        class: total_increasing
+        mapping:
+          - scale: 100
+  #     - id: 131
+  #       type: integer
+  #       name: calibration
+  #       step: 1
+  #       range:
+  #         min: 800
+  #         max: 1200
+  #       mapping:
+  #         - scale: 1000
+  - entity: sensor
+    name: Reverse energy A
+    class: energy
+    category: diagnostic
+    dps:
+      - id: 108
+        type: integer
+        name: sensor
+        unit: kWh
+        class: total_increasing
+        mapping:
+          - scale: 100
+  #     - id: 132
+  #       type: integer
+  #       name: calibration
+  #       step: 1
+  #       range:
+  #         min: 800
+  #         max: 1200
+  #       mapping:
+  #         - scale: 1000
+  - entity: sensor
+    name: Forward energy A total
```

Comment:
It isn't clear what the difference is between "Forward energy A" and "Forward energy A total", but one should probably be merged to an attribute of the other along with the hourly one. The one that resets the least is probably going to be the best one to use for HA.